src/lib.rs

```diff
@@ -1,71 +1,59 @@
-of using `local_ip` directly.
+println!("This is my local IP address: {:?}", my_local_ip);
+```
 
-Help improve the support for multiple systems on this crate by opening a pull
-request or issue on [GitHub](https://github.com/EstebanBorai/local-ip-address).
+Retrieve all the available network interfaces from both, the `AF_INET` and
+the `AF_INET6` family by executing the `list_afinet_netifas` function:
 
-```ignore
-use std::net::IpAddr;
-use local_ip_address::find_af_inet;
+```rust
+use local_ip_address::list_afinet_netifas;
 
-let ifas = find_af_inet().unwrap();
+let network_interfaces = list_afinet_netifas().unwrap();
 
-if let Some((_, ipaddr)) = ifas
-    .iter()
-    .find(|(name, ipaddr)| *name == "en0" && matches!(ipaddr, IpAddr::V4(_))) {
-    println!("This is your local IP address: {:?}", ipaddr);
-    // This is your local IP address: 192.168.1.111
-    assert!(matches!(ipaddr, IpAddr));
+for (name, ip) in network_interfaces.iter() {
+    println!("{}:\t{:?}", name, ip);
 }
 ```
 
+Underlying approach on retrieving network interfaces or the local IP address
+may differ based on the running operative system.
+
+OS | Approach
+--- | ---
+Linux | Establishes a Netlink socket interchange to retrieve network interfaces
+macOS | Uses of `getifaddrs` to retrieve network interfaces
+Windows | Consumes Win32 API's to retrieve the network adapters table
+
 */
 use std::net::IpAddr;
-use std::string::FromUtf8Error;
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    /// An error occured building a `&str` from a C string when
-    /// parsing the name of a interface address instance
-    #[error("Failed to read interface address name. `{0}`")]
-    IntAddrNameParseError(FromUtf8Error),
-    /// An error ocurred calling `getifaddrs`
-    #[error("Execution of getifaddrs had error result. getifaddrs returned `{0}`")]
-    GetIfAddrsError(i32),
-    /// The current platform is not supported
-    #[error("The current platform `{0}` is not supported")]
+    /// Returned when `local_ip` is unable to find the system's local IP address
+    /// in the collection of network interfaces
+    #[error("The Local IP Address wasn't available in the network interfaces list/table")]
+    LocalIpAddressNotFound,
+    /// Returned when an error occurs in the strategy level.
+    /// The error message may include any internal strategy error if available
+    #[error("An error ocurred executing the underlying strategy error.\n{0}")]
+    StrategyError(String),
+    /// Returned when the current platform is not yet supported
+    #[error("The current platform: `{0}`, is not suppported")]
     PlatformNotSupported(String),
-    #[error("GetIpAddrTableError")]
-    GetAdaptersAddresses(u32),
-    #[error("Netlink IO Error `{0}`")]
-    NetlinkIOError(String),
-    #[error("Netlink: Socket Message Error `{0}`")]
-    NetlinkSendMessageError(String),
-    #[error("Netlink: Failed to find local IP address")]
-    NetlinkFailedToFindLocalIp,
```

Comment:
Dedicated errors have been removed in place of generic errors compatible with multiple implementations